tests/packages/cmake_defines/CMakePresets.json

```diff
@@ -0,0 +1,14 @@
+{
+  "version": 1,
+  "configurePresets": [
+    {
+      "name": "scikit",
+      "cacheVariables": {
+        "PRESET_ONLY_VAR": "defined",
+        "OVERWRITTEN_VAR": "original"
+      },
+      "binaryDir": "/dev/null",
+      "generator": "Ninja"
```

Comment:
Windows disapproves, but empty generator is only available in version `3`. Not sure if I can bump it and satisfy the other builders. What generator value is available on windows?

Reply:
Windows is the black sheep here, and there is some logic on scikit-build-core around it. Don't remember the details though. There is also the logic around `make-fallback` option which confused me a bunch when I first attempted this